bundles/AdminBundle/Controller/Admin/NotificationController.php

```diff
@@ -0,0 +1,222 @@
+<?php
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Enterprise License (PEL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ * @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ * @license    http://www.pimcore.org/license     GPLv3 and PEL
+ */
+
+declare(strict_types=1);
+
+namespace Pimcore\Bundle\AdminBundle\Controller\Admin;
+
+use Pimcore\Bundle\AdminBundle\Controller\AdminController;
+use Pimcore\Bundle\AdminBundle\Helper\QueryParams;
+use Pimcore\Bundle\AdminBundle\HttpFoundation\JsonResponse;
+use Pimcore\Model\Element\Service;
+use Pimcore\Model\Notification\Service\NotificationService;
+use Pimcore\Model\Notification\Service\NotificationServiceFilterParser;
+use Pimcore\Model\Notification\Service\UserService;
+use Pimcore\Model\User;
+use Pimcore\Translation\Translator;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Method;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Routing\Annotation\Route;
+
+/**
+ * @Route("/notification")
+ */
+class NotificationController extends AdminController
+{
+    /**
+     * @param UserService $service
+     * @return JsonResponse
+     * @Route("/recipients")
+     * @Method({"GET"})
+     */
+    public function recipientsAction(UserService $service, Translator $translator): JsonResponse
+    {
+        $this->checkPermission('notifications');
```

Comment:
notifications permission is not installed properly: 
- missing migration (same goes also for notifications table) 

in general, I would add two permissions:
- list/receive permissions
- send permissions

Reply:
I added new permission:
notifications_send

 I added permissions in file `/bundles/InstallBundle/Installer.php:718` in $userPermissions array so I think permissions will insert to database.